src/core/config-schema.helper.ts

```diff
@@ -0,0 +1,125 @@
+import * as Joi from '@hapi/joi';
+import chalk from 'chalk';
+
+const schema = Joi.object({
+  browserWidth: Joi.number()
+    .min(100)
+    .max(2000)
+    .required(),
+  browserHeight: Joi.number()
+    .min(100)
+    .max(2000)
+    .required(),
+  browserLanguage: Joi.string().required(),
+  timeout: Joi.number()
+    .min(1)
+    .required(),
+  elementsVisibilityTimeout: Joi.number()
+    .min(1)
+    .required(),
+  waitForPageTimeout: Joi.number()
+    .min(1)
+    .required(),
+  downloadTimeout: Joi.number()
+    .min(1)
+    .required(),
+  reports: Joi.string().required(),
+  downloads: Joi.string().required(),
+  data: Joi.string().required(),
+  features: Joi.array()
+    .items(Joi.string())
+    .required(),
+  pages: Joi.array()
+    .items(Joi.string())
+    .required(),
+  matchers: Joi.array()
+    .items(Joi.string())
+    .required(),
+  generators: Joi.array()
```

Comment:
Just a quick question, in case if someone doesn't want to use generators, form handlers dictionaries transformators etc. ? In case you present it is required even if I don't use it to create a path and enter it inside kakunin.conf
I would leave it optional

Reply:
I've verified how should look like the minimum config for running tests:

```
module.exports = {
  timeout: 60,
  reports: '/reports',
  features: ['/features'],
  pages: ['/pages'],
  step_definitions: ['/step_definitions'],
  imageComparator: {
    baselineFolder: 'baseline',
  },
  type: 'otherWeb',
  baseUrl: 'http://localhost:8080',
};
```
Add ```required()``` methods on those fields. 

I wouldn't use ```optional()``` method because it could accept undefined.

Reply:
I just wonder why `imageComparator` is required, it's functionality like `emailService`, `generators` etc. @sethii, @tgorskievo  should it really be required ?

Reply:
@simonsipl we use plugin in `protractor.conf.js` I could try to set default value for baseline folder but do we need it? @sethii @tgorskievo